controllers/che/backup.go

```diff
@@ -0,0 +1,119 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package che
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	semver "github.com/blang/semver/v4"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	chev1 "github.com/eclipse-che/che-operator/api/v1"
+	"github.com/eclipse-che/che-operator/pkg/deploy"
+)
+
+const (
+	DefaultBackupServerConfigLabelKey = "che.eclipse.org/default-backup-server-configuration"
+)
+
+func isCheGoingToBeUpdated(cheCR *chev1.CheCluster) bool {
+	deployedCheVersion := cheCR.Status.CheVersion
+	newCheVersion := deploy.DefaultCheVersion()
+	if deployedCheVersion == "" || deployedCheVersion == "next" || deployedCheVersion == newCheVersion {
+		return false
+	}
+
+	deployedSemver, err := semver.Make(deployedCheVersion)
+	if err != nil {
+		logrus.Warn(getSemverParseErrorMessage(deployedCheVersion, err))
+		return false
+	}
+	currentSemver, err := semver.Make(newCheVersion)
+	if err != nil {
+		logrus.Warn(getSemverParseErrorMessage(newCheVersion, err))
+		return false
+	}
+	return currentSemver.GT(deployedSemver)
+}
+
+func getSemverParseErrorMessage(version string, err error) string {
+	return fmt.Sprintf("It is not possible to parse a current version '%s'. Cause: %v", version, err)
+}
+
+func getBackupCR(deployContext *deploy.DeployContext) (*chev1.CheClusterBackup, error) {
```

Comment:
```suggestion
func getBackupCRForCurrentCheVersion(deployContext *deploy.DeployContext) (*chev1.CheClusterBackup, error) {
```

Reply:
I don't think it may any sense to name the function like this. It should just fetch backup CR. Which version it belongs to - is another question.

Reply:
I'd use `getBackupCRForUpdate` if you don't mind.